client/src/partials/purchases/create/PurchaseOrderForm.js

```diff
@@ -0,0 +1,311 @@
+    }
+
+    // bind the cache key
+    this.cache = AppCache(cacheKey);
+
+    // set up the inventory
+    // this will be referred to as PurchaseOrderForm.inventory.available.data
+    this.inventory = new Pool({ identifier: 'uuid', data : [] });
+
+    // set up the inventory
+    // @todo - if necessary, we could call this in a reload() or setup() method
+    Inventory.getInventoryItems()
+      .then(function (data) {
+        this.inventory.initialize('uuid', data);
+      }.bind(this));
+
+    // setup the rows of the grid as a store
+    // this will be referred to as PurchaseOrderForm.store.data
+    this.store = new Store({ identifier : 'uuid', data: [] });
+
+    this.setup();
+  }
+
+  // initial setup and clearing of the order
+  PurchaseOrderForm.prototype.setup = function setup() {
+
+    // the order details
+    this.details = {
+      payment_method: 'FORM.LABELS.ON_DELIVERY',
+      date: new Date(),
+      cost: 0,
+    };
+
+    // the supplier is null
+    this.supplier = null;
+
+    // this object holds the totals for the order.
+    this.totals = {
+      rows : 0
+    };
+
+    // remove all items from the store as needed
+    this.clear();
+
+    this._valid = false ;
+    this._invalid = true;
+
+    // trigger a totals digest
+    this.digest();
+  };
+
+  /**
+   * @method validate
+   *
+   * @description
+   * This method digests the order, then returns all invalid items in the
+   * order to be dealt with by the user.
+   */
+  PurchaseOrderForm.prototype.validate = function validate() {
+    this.digest();
+
+    // filters out valid items
+    var invalidItems = this.store.data.filter(function (row) {
+      return row._invalid;
+    });
+
+    this._invalid = invalidItems.length > 0;
+    this._valid = !this._invalid;
+
+    return invalidItems;
+  };
+
+
+  /**
+   * @method setSupplier
+   *
+   * @description
+   * This allows the supplier to be cached locally and hooks up the creditor
+   * uuid to the details form.
+   *
+   * @param {Object} supplier- a supplier selected by the purchasing
+   *   module's typeahead
+   */
+  PurchaseOrderForm.prototype.setSupplier = function setSupplier(supplier) {
+    var order = this;
+
+    // attach the creditor uuid to the request
+    order.details.supplier_uuid = supplier.uuid;
+
+    // add a single item to the order to begin
+    if (order.store.data.length === 0) {
+      order.addItem();
+    }
+
+    // run validation and calculation
+    order.digest();
+
+  };
+
+  /**
+   * @method digest
+   *
+   * @description
+   * Calculates the totals for the order by summing all the values in the grid.
+   *
+   * This method should be called anytime the values of the grid change, and on
+   * setSupplier() completion.
+   */
+  PurchaseOrderForm.prototype.digest = function digest() {
+    var order = this;
+    var totals = this.totals;
+
+    // loop through the items summing them into a total
+    totals.rows = this.store.data.reduce(function (value, unit) {
```

Comment:
It would probably be good to consistently either use the `var order = this;` alias to always use `this`, mixing the two can be a bit confusing reading through the code. 

The functionality will remain exactly the same but I think it would be good use `order` after referencing it here.